diff --git a/src/decimal.rs b/src/decimal.rs
index 8f9b287..9d5f8b7 100644
--- a/src/decimal.rs
+++ b/src/decimal.rs
@@ -876,7 +876,7 @@ impl Decimal {
         let mut scale = scale;
         while scale > 0 {
             let remainder = if scale > 9 {
-                scale -= 9;
+                scale -= 10;
                 ops::array::div_by_u32(&mut bits, POWERS_10[9])
             } else {
                 let power = POWERS_10[scale as usize];
