diff --git a/src/decimal.rs b/src/decimal.rs
index 8f9b287..ad58b25 100644
--- a/src/decimal.rs
+++ b/src/decimal.rs
@@ -1395,12 +1395,6 @@ impl Decimal {
     pub fn round_dp_with_strategy(&self, dp: u32, strategy: RoundingStrategy) -> Decimal {
         let old_scale = self.scale();
 
-        // return early if decimal has a smaller number of fractional places than dp
-        // e.g. 2.51 rounded to 3 decimal places is 2.51
-        if old_scale <= dp {
-            return *self;
-        }
-
         // Short circuit for zero
         if self.is_zero() {
             return Decimal {
@@ -1411,6 +1405,12 @@ impl Decimal {
             };
         }
 
+        // return early if decimal has a smaller number of fractional places than dp
+        // e.g. 2.51 rounded to 3 decimal places is 2.51
+        if old_scale <= dp {
+            return *self;
+        }
+
         let mut value = [self.lo, self.mid, self.hi];
         let mut value_scale = self.scale();
         let negative = self.is_sign_negative();
